nixos/modules/services/networking/onedrive.nix

```diff
@@ -0,0 +1,88 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.onedrive;
+
+  onedriveLauncher =  pkgs.writeShellScriptBin
+    "onedrive-launcher"
+    ''
+      # XDG_CONFIG_HOME is not recognized in the environment here.
+      if [ -f $HOME/.config/onedrive-launcher ]
+      then
+        # Hopefully using underscore boundary helps locate variables
+        for _onedrive_config_dirname_ in $(cat $HOME/.config/onedrive-launcher | grep -v '[ \t]*#' )
+        do
+          systemctl --user start onedrive@$_onedrive_config_dirname_
+        done
+      else
+        systemctl --user start onedrive@onedrive
+      fi
+    ''
+  ;
+
+  # Poor man's type/typo safety
+  systemdTargetList = [
+    "network-online"  
+    "multi-user"
+  ];
+
+  systemdTargets = builtins.listToAttrs
+  (
+    map
+      ( st:{ name = st; value = st + ".target" ; } )
+      systemdTargetList
+  );
+
+
+in {
+  ### Documentation
+  meta.doc = ./onedrive.xml;
+
+  ### Interface 
+
+  options.services.onedrive = {
+    enable = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = "Enable OneDrive service";
+    };
+
+     package = lib.mkOption {
```

Comment:
Do we have multiple onedrive packages to choose from or does the single package have a lot of compile time flags that people would want to set? If not, get rid of this.

Reply:
I've left this in because onedrive upstream moves very fast, and there might be times where one is subscribed to a release channel but would like to use the latest package. For example in fact reportedly the version in stable has problems with onedrive business, and this pr-  https://github.com/NixOS/nixpkgs/pull/89198 upgrades the version. 

(I know it's possible to use a specific service from a specific channel/commit but that's a tiny bit more work than using a package from another channel)